pkg/asset/rhcos/image.go

```diff
@@ -93,20 +93,37 @@ func osImage(ctx context.Context, config *types.InstallConfig) (string, string,
 		if ami := config.Platform.AWS.AMIID; len(ami) > 0 {
 			return ami, ami, nil
 		}
-		region := config.Platform.AWS.Region
+		const globalResourceRegion = "us-east-1"
+		var osimageControlPlane, osimageCompute string
+		region := config.AWS.Region
 		if !rhcos.AMIRegions(config.ControlPlane.Architecture).Has(region) {
-			const globalResourceRegion = "us-east-1"
 			logrus.Debugf("No AMI found in %s. Using AMI from %s.", region, globalResourceRegion)
 			region = globalResourceRegion
 		}
-		osimageControlPlane, err := st.GetAMI(archName, region)
+		osimageControlPlane, err = st.GetAMI(archName, region)
 		if err != nil {
 			return "", "", err
 		}
 		if region != config.Platform.AWS.Region {
 			osimageControlPlane = fmt.Sprintf("%s,%s", osimageControlPlane, region)
 		}
-		return osimageControlPlane, osimageControlPlane, nil
```

Comment:
I would rather that osImage function takes arch as a parameter and pass in the architecture through the generate call  rather than doing this. it makes it more readable and clean.

Reply:
this way it becomes generic for all platforms and not just aws

Reply:
Can you exemplify what you mean?

Reply:
refer to : https://github.com/Prashanth684/installer/commit/49582c8174facbf1650302445fa76e91569254c4#diff-da71df8f5aaa245588aad9ad03eac91400cfa2afe2072f3eb04aa6125535e85dR70

Reply:
Ah ok, got it.

Reply:
Done.